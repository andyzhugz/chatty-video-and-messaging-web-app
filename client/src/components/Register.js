import React, { useState, useEffect } from 'react'
import {Link, useHistory} from 'react-router-dom'
import axios from "axios"; 
import { register } from '../javascripts/authRequests'
import '../stylesheets/LoginRegister.css'

function Register(user) {
    const history=useHistory()
    const [username, setUsername] = useState("");
    const [nickname, setNickname] = useState("");
    const [password, setPassword] = useState("");
    const [isSubmitting, setIsSubmitting] = useState(false);

    const handleSubmit = () => {
        const userName = document.getElementById('userName').value;
        const nickName = document.getElementById('nickName').value;
        const pwd = document.getElementById('password').value;
        setIsSubmitting(true);
        // leave axios code here
        register(userName, nickName, pwd);
        setIsSubmitting(false);
    };

    return  (
        <div>
            <h1 align="center">Welcome to Our App</h1>
            <div className="main">
                <p className="sign" align="center">Register</p>
                <form className="formLogin">
                    <input className="userName" id="userName" type="text" align="center" placeholder="Username: cat@gmail.com" />
                    <input className="nickName" id="nickName" type="text" align="center" placeholder="Nickname" />
                    <input className="pass" id="password" type="password" align="center" placeholder="Password" />
                    <input type="button" id="submitBtn" value={isSubmitting ? 'Submitting' : 'Register'} onClick={handleSubmit} className="registerBtn2" />
                </form>  
            </div>
        </div>
    );
}

export default Register;