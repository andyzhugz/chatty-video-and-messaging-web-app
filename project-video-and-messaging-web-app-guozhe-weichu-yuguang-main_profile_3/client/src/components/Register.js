import React, { useState, useEffect } from 'react'
import {Link,useHistory}from 'react-router-dom'
import axios from "axios"; 
import {register} from '../javascripts/authRequests';
import '../stylesheets/LoginRegister.css'

function Register(user) {
    const history=useHistory()
    const [username, setUsername] = useState("");
    const [nickname, setNickname] = useState("");
    const [password, setPassword] = useState("");
    const [isSubmitting, setIsSubmitting] = useState(false);

    const handleSubmit = () => {
        const userName = document.getElementById('userName').value;
        const nickName = document.getElementById('nickName').value;
        const pwd = document.getElementById('password').value;
        setIsSubmitting(true);
        register(userName,nickName,pwd);
        // const requestOptions = {
        //     method: 'POST',
        //     headers: {
        //       'Content-Type': 'application/json',
        //       Accept: 'application/json',
        //     },
        //     body: JSON.stringify({ email: userName, username: nickName, password: pwd }),
        //   };
        // fetch('http://localhost:8080/Register', requestOptions)
        // .then((back) => console.log(back))
        // .catch(error => {
        //     console.log(error);
        //     setIsSubmitting(false);

          // wait for backend
          // if (error.response.status === 401) setError(error.response.data.message);
          // else 
        //});
    };
    
    return  (
        <div>
            <h1 align="center">Welcome to Our App</h1>
            <div className="main">
                <p className="sign" align="center">Register</p>
                <form className="formLogin">
                    <input className="userName" id="userName" type="text" align="center" placeholder="Username: cat@gmail.com" />
                    <input className="nickName" id="nickName" type="text" align="center" placeholder="Nickname" />
                    <input className="pass" id="password" type="password" align="center" placeholder="Password" />
                    <input type="button" value={isSubmitting ? 'Submitting' : 'Register'} onClick={handleSubmit} className="registerBtn2" />
                </form>  
            </div>
        </div>
        );
}

export default Register;