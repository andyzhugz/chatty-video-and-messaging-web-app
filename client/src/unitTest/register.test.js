import React from 'react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import { Router } from 'react-router-dom';
import Register from "../components/Register.js";

const renderer = require('react-test-renderer');

// snapshot testings
describe('Test snapshot', () => {
  test('Test Login', () => {
    const component = renderer.create(<Register />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// functonal component functions tests
describe('Test functonal component functions', () => {
    let container;
    beforeEach(() => {
      container = document.createElement('div');
      document.body.appendChild(container);
    });
  
    afterEach(() => {
      document.body.removeChild(container);
      container = null;
    });
  
    test('Test function inside Register component', () => {
      act(() => {
            ReactDOM.render(<Register />, container);
      });
      const submitBtn = container.querySelector('#submitBtn');
      act(() => {
        fetch = jest.fn(() => Promise.resolve({ json: () => ({ message: 'success', data: [{ username: 'tester', password: 'tester' }] }) }));
        submitBtn.dispatchEvent(new MouseEvent('click', { bubbles: true }));
      });
      const userName = container.querySelector('#userName');
      const nickName = container.querySelector('#nickName');
      const password = container.querySelector('#password');
      expect(userName.value).toBe("");
      expect(nickName.value).toBe("");
      expect(password.value).toBe("");
    });
  
    test('Test function inside Login component failure', async() => {
        act(() => {
          ReactDOM.render(<Register />, container);
        });
        const submitBtn = container.querySelector('#submitBtn');
        await act(async () => {
          fetch = jest.fn(() => Promise.reject({ json: () => ({ message: 'failed', data: { player: 'tester', point: 2, id: 1 }})}));
          submitBtn.dispatchEvent(new MouseEvent('click', { bubbles: true }));
        });
      });
  });
  
