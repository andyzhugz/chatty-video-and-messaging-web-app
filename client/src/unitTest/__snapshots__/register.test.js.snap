// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Test Login 1`] = `
<div>
  <h1
    align="center"
  >
    Welcome to Our App
  </h1>
  <div
    className="main"
  >
    <p
      align="center"
      className="sign"
    >
      Register
    </p>
    <form
      className="formLogin"
    >
      <input
        align="center"
        className="userName"
        id="email"
        placeholder="Email: cat@gmail.com"
        type="text"
      />
      <input
        align="center"
        className="nickName"
        id="userName"
        placeholder="Username"
        type="text"
      />
      <input
        align="center"
        className="pass"
        id="password"
        placeholder="Password"
        type="password"
      />
      <input
        className="registerBtn2"
        id="submitBtn"
        onClick={[Function]}
        type="button"
        value="Register"
      />
    </form>
  </div>
</div>
`;
